vector_getters.go

```diff
@@ -201,3 +201,49 @@ func (vec *vector) getSliceChild(offset uint64, length uint64) []any {
 	}
 	return slice
 }
+
+func (vec *vector) getUnion(rowIdx mapping.IdxT) any {
+	if vec.getNull(rowIdx) {
+		return Union{Tag: "", Value: nil}
+	}
+
+	// Get the tag vector (stored at index 0)
+	tagVector := mapping.StructVectorGetChild(vec.vec, 0)
```

Comment:
For performance, maybe instead of retrieving the tag vector on every row that's scanned, we should cache it as part of the child vectors? Then, we can also cache its data pointer, instead of calling `VectorGetData` on each row being scanned.

Reply:
That sounds like a good idea. I tried implementing it and the tests are passing.
I added `tagDataPtr unsafe.Pointer` to `vectorTypeInfo` and I updated `getUnion` and `getChildVectors`. Would be great if you can check if I am missing anything. Thanks.